Custom Sound Example in Wiki.

Phoenixvine32908 opened the pull request:
## What
Adds a custom sound example for the wiki. Also shows the datagen necessary to make sounds.json.

## Outcome
Allows making custom sounds easier for addon/packdevs.

## Potential Compatibility Issues
Shouldn't be any. (Sorry for the comment history jurre, I'll fix it eventually)

Phoenixvine32908 committed:
Make new flag DISABLE_MATERIAL_RECIPES to replace NO_UNIFICATION.

Phoenixvine32908 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1

# Conflicts:
#	src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/Material.java
#	src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/info/MaterialFlags.java
#	src/main/java/com/gregtechceu/gtceu/common/data/GTRecipes.java

Phoenixvine32908 committed:
Recipe conditon docu update.

Phoenixvine32908 committed:
Merge branch '1.20.1' into 1.20.1

Ghostipedia committed:
Merge branch '1.20.1' into 1.20.1

Phoenixvine32908 committed:
Update to 7.2.1

Phoenixvine32908 committed:
Merge branch '1.20.1' into 1.20.1

Phoenixvine32908 committed:
Update docs/content/Modpacks/Recipes/Recipe-Conditions.md

Co-authored-by: Jurre Groenendijk <jurre@jilles.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Recipes/Recipe-Conditions.md

Co-authored-by: Jurre Groenendijk <jurre@jilles.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Recipes/Recipe-Conditions.md

Co-authored-by: Jurre Groenendijk <jurre@jilles.com>

Phoenixvine32908 committed:
Jurre requested fixes. As well as fixing the link.

Phoenixvine32908 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1

Phoenixvine32908 committed:
JurrThese fix.

Phoenixvine32908 committed:
Update docs/content/Modpacks/Examples/Custom-Recipe-Condition.md

Co-authored-by: Jurre Groenendijk <jurre@jilles.com>

Phoenixvine32908 committed:
Indentation.

Phoenixvine32908 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1

Phoenixvine32908 committed:
Removal of rpm recipe type and fixing the apperance block of example turbine and steam multiblocks.

Phoenixvine32908 committed:
Merge branch 'GregTechCEu:1.20.1' into 1.20.1

Phoenixvine32908 committed:
Merge branch 'GregTechCEu:1.20.1' into 1.20.1

Phoenixvine32908 committed:
Custom sound example.

Phoenixvine32908 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1

Phoenixvine32908 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
```diff
@@ -0,0 +1,48 @@
+## Creating a custom sound
+
+!!! Warning
+    Registering custom sounds is currently only supported in Java. Although you can call your sound in kubejs once defined.
```
```suggestion
    Registering custom sounds is currently only supported in Java, though you can use your sound in kubejs scripts once it's defined.
```

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
```diff
@@ -0,0 +1,48 @@
+!!! Warning
+    Registering custom sounds is currently only supported in Java. Although you can call your sound in kubejs once defined.
+
+To add a new sound, a sounds class is required. 
```
###### well, not exactly... It's just recommended to put them in a separate class

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
````diff
@@ -0,0 +1,48 @@
+An example of a custom sound can be found below.
+
+```java
+import static com.gregtechceu.gtceu.common.registry.GTRegistration.REGISTRATE;
````
Do. Not. Use. GT's registrate.
Defining sounds requires a bit of datagen which **will not run** for your mod if you use GT's registrate instance.
It'd also register them to the wrong nsmespace.

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
```diff
@@ -0,0 +1,48 @@
+
+public class ExampleSound {
+
+    public static final SoundEntry MICROVERSE = REGISTRATE.sound(examplemod.id("microverse")).build();
```
```suggestion
    public static final SoundEntry MICROVERSE = REGISTRATE.sound("microverse").build();
```

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
````diff
@@ -0,0 +1,48 @@
+}
+```
+
+After you make this class and init it inside your main mod class, you want to setup datagen for the sound.
````
```suggestion
After you make this class and initialize it in your main mod class, you want to setup datagen for the sounds.
It's a bit more complicated than normal datagen, so an example can be found below.
```
If I didn't manage to select the 2nd line (am on mobile) please do the change manually

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
````diff
@@ -0,0 +1,48 @@
+It's a bit more complicated than normal datagen so an example can be found below.
+
+```java
+@Mod.EventBusSubscriber(modid = examplemod.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
````
```suggestion
@Mod.EventBusSubscriber(modid = ExampleMod.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
```

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
You should also describe how to actually get the data to generate properly as it's not obvious

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
````diff
@@ -0,0 +1,48 @@
+
+```
+
+Finally, you need to actually prepare your sound for use. For a sound to be registered it must be in .ogg format and be inside assets/examplemod/sounds. 
````
This has to be done before running datagen or it'll fail

Phoenixvine32908 replied:
Redid it a bit, lemme know if it still needs some refining.

screret commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
````diff
@@ -0,0 +1,48 @@
+```
+
+Finally, you need to actually prepare your sound for use. For a sound to be registered it must be in .ogg format and be inside assets/examplemod/sounds. 
+Mono sounds are localized to the area around the machine, while stereo sounds are global.
````
```suggestion
!!! note "mono vs. stereo audio"

    Your audio file should be mono, as Minecraft's attentuation logic only works with single-channel audio. Stereo sounds won't fade out (they'll be played at the same volume at any distance from the source) and should only be used for background tracks such as the main menu music. 
```

screret reviewed (changes requested):
You should also describe how to actually get the data to generate properly as it's not obvious

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: screret <68943070+screret@users.noreply.github.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: screret <68943070+screret@users.noreply.github.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: screret <68943070+screret@users.noreply.github.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: screret <68943070+screret@users.noreply.github.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: screret <68943070+screret@users.noreply.github.com>

Phoenixvine32908 committed:
Revise custom sound registration guidelines

Updated sound registration instructions and clarified audio format requirements.

Phoenixvine32908 committed:
Update Custom-Sounds.md

Phoenixvine32908 commented:
Updated a few things, I'll try to fix the rest once I get back from work.

Phoenixvine32908 committed:
Merge branch '1.20.1' into 1.20.1

jurrejelle commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
````diff
@@ -0,0 +1,50 @@
+}
+```
+
+Before you run datagen, the sound needs to be prepared for use. For a sound to be registered it must be in .ogg format and be inside assets/examplemod/sounds. 
````
```suggestion
Before you run datagen, the sound needs to be prepared for use. For a sound to be registered it must be in .ogg format and be inside `assets/examplemod/sounds`. 
```

jurrejelle commented on docs/content/Modpacks/Other-Topics/Custom-Sounds.md:
```diff
@@ -0,0 +1,50 @@
+---
+
+
+## Creating a custom sound
```
```suggestion
## Creating a Custom Sound
```

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: Jurre Groenendijk <jurre@jilles.com>

Phoenixvine32908 committed:
Update docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Co-authored-by: Jurre Groenendijk <jurre@jilles.com>

Phoenixvine32908 committed:
Merge branch '1.20.1' into 1.20.1
